### PR TITLE
Import DataFusion Rust runner from datafusion-contrib

### PR DESCRIPTION
**Pull request by andygrove**

### CODE REVIEW

File: runners/datafusion-rust/src/main.rs [hunk 1]
```diff
@@ -0,0 +1,291 @@
...
+            query_times: vec![],
+        }
+    }
+}
+
+#[tokio::main]
+pub async fn main() -> Result<()> {
+    let mut results = Results::new();
+    for arg in std::env::args() {
+        results.command_line_args.push(arg);
```
Comment (viirya) [inline runners/datafusion-rust/src/main.rs:94]: Do we use this `command_line_args` later?

File: runners/datafusion-rust/src/main.rs [hunk 2]
```diff
@@ -0,0 +1,291 @@
...
+
+    /// Number of queries in this benchmark suite
+    #[structopt(short, long)]
+    num_queries: Option<u8>,
+
+    /// List of queries to exclude
+    #[structopt(short, long)]
+    exclude: Vec<u8>,
+
+    /// Concurrency
```
Comment (viirya) [inline runners/datafusion-rust/src/main.rs:54]:
```suggestion
    /// Concurrency, determining the number of partitions for queries
```

File: runners/datafusion-rust/src/main.rs [hunk 3]
```diff
@@ -0,0 +1,291 @@
...
+                } else {
+                    let filename = format!("{}/q{}{}.csv", output_path, query_no, file_suffix);
+                    let t = MemTable::try_new(batches[0].schema(), vec![batches])?;
+                    let df = ctx.read_table(Arc::new(t))?;
+                    df.write_csv(&filename, DataFrameWriteOptions::default(), None)
+                        .await?;
+                }
+            }
+        }
+        durations.push(total_duration_millis);
```
Comment (viirya) [inline runners/datafusion-rust/src/main.rs:287]: Looks like this takes total duration of all iterations. I'm wondering if we should use average duration?

File: runners/datafusion-rust/src/main.rs (hunk 1 above)
Comment (andygrove) [reply runners/datafusion-rust/src/main.rs:94]: Yes, the `results` struct gets written out as json. We store the CLI args so we can see exactly how the benchmark was run.

```rust
let json = serde_json::to_string_pretty(&results).unwrap();
```

File: runners/datafusion-rust/src/main.rs (hunk 3 above)
Comment (andygrove) [reply runners/datafusion-rust/src/main.rs:287]: I think this is just hard to parse from looking at the diff here, but the code is:

```rust
    for iteration in 0..iterations {
        // duration for executing all queries in the file
        let mut total_duration_millis = 0;
        for (i, sql) in sql.iter().enumerate() {
            ...
        }
        durations.push(total_duration_millis);
    }
```

Comment (andygrove) [reply runners/datafusion-rust/src/main.rs:287]: `total_duration_millis` is the total duration for all queries in the file (some query files have multiple queries)

**Review by viirya — APPROVED** [review]

**Comment by viirya** [comment]
Looks good to me. Thanks @andygrove.
